components/brave_wallet/browser/simple_hash_client.cc

```diff
@@ -591,6 +862,314 @@ SimpleHashClient::ParseNFTsFromSimpleHash(const base::Value& json_value,
+  if (!dict) {
+    return std::nullopt;
+  }
+
+  base::flat_map<std::string, std::string> token_urls;
+  const base::Value::List* nfts = dict->FindList("nfts");
+  if (!nfts) {
+    return std::nullopt;
+  }
+
+  // For each NFT, create a composite key of chain_id.contract_address or
+  // chain_id.contract_address.token_id if token_id is present.
+  for (const auto& nft_value : *nfts) {
+    const base::Value::Dict* nft = nft_value.GetIfDict();
+    if (!nft) {
+      continue;
+    }
+
+    const std::string* chain_id = nft->FindString("chain");
+    if (!chain_id) {
+      continue;
+    }
+
+    const std::string* contract_address = nft->FindString("contract_address");
+    if (!contract_address) {
+      continue;
+    }
+
+    const std::string* token_id = nft->FindString("token_id");
+
+    const base::Value::Dict* extra_metadata = nft->FindDict("extra_metadata");
+    if (!extra_metadata) {
+      continue;
+    }
+
+    const std::string* metadata_original_url =
+        extra_metadata->FindString("metadata_original_url");
+    if (!metadata_original_url) {
+      continue;
+    }
+
+    std::optional<std::string> chain_id_str =
+        SimpleHashChainIdToChainId(*chain_id);
+    if (!chain_id_str) {
+      continue;
+    }
+
+    // Create the composite key and convert to lowercase using
+    // base::ToLowerASCII.
+    std::string composite_key =
+        base::StrCat({*chain_id_str, ".", *contract_address});
+    if (token_id && !token_id->empty()) {
+      composite_key += "." + *token_id;
+    }
+    composite_key = base::ToLowerASCII(composite_key);
+
+    token_urls[composite_key] = *metadata_original_url;
+  }
+
+  return token_urls;
+}
+
+std::optional<base::flat_map<std::string, mojom::NftMetadataPtr>>
+SimpleHashClient::ParseMetadatas(const base::Value& json_value) {
+  const base::Value::Dict* dict = json_value.GetIfDict();
+  if (!dict) {
+    return std::nullopt;
+  }
+
+  const base::Value::List* nfts = dict->FindList("nfts");
+  if (!nfts) {
+    return std::nullopt;
+  }
+
+  base::flat_map<std::string, mojom::NftMetadataPtr> nft_metadatas;
+  for (const auto& nft_value : *nfts) {
+    const base::Value::Dict* nft = nft_value.GetIfDict();
+    if (!nft) {
+      continue;
+    }
+
+    const std::string* chain_id = nft->FindString("chain");
+    if (!chain_id) {
+      continue;
+    }
+
+    const std::string* contract_address = nft->FindString("contract_address");
+    if (!contract_address) {
+      continue;
+    }
+
+    const std::string* token_id = nft->FindString("token_id");
+
+    std::optional<std::string> chain_id_str =
+        SimpleHashChainIdToChainId(*chain_id);
+    if (!chain_id_str) {
+      continue;
+    }
+
+    std::string composite_key =
+        base::StrCat({*chain_id_str, ".", *contract_address});
+    if (token_id && !token_id->empty()) {
+      composite_key += "." + *token_id;
+    }
+    composite_key = base::ToLowerASCII(composite_key);
+
```

Comment:
If we still need string composite_key then it should be an one anonymous function for building it
```
std::string MakeCompositeKey(const std::string& chain_id_str, const std::string& contract_address, const std::string* token_id ) {
   std::string composite_key =
        base::StrCat({*chain_id_str, ".", *contract_address});
    if (token_id && !token_id->empty()) {
      composite_key += "." + *token_id;
    }
    return base::ToLowerASCII(composite_key);
}
```